src/main/java/org/fundacionjala/virtualassistant/player/spotify/client/SpotifyClient.java

```diff
@@ -0,0 +1,364 @@
+        try {
+            HttpHeaders headers = new HttpHeaders();
+            headers.set("Authorization", "Bearer " + accessToken);
+            headers.setContentType(MediaType.APPLICATION_JSON);
+
+            String requestBody = "{\"uris\": [\"" + trackUri + "\"]}";
+            HttpEntity<String> entity = new HttpEntity<>(requestBody, headers);
+
+            RestTemplate restTemplate = new RestTemplate();
+            ResponseEntity<String> response = restTemplate.exchange(
+                    "https://api.spotify.com/v1/me/player/play",
+                    HttpMethod.PUT,
+                    entity,
+                    String.class
+            );
+
+            return response.getStatusCode() == HttpStatus.NO_CONTENT;
+        } catch (Exception e) {
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+    public String getAccessToken() {
+        return accessToken;
+    }
+
+    public void setAccessToken(String accessToken) {
+        this.accessToken = accessToken;
+    }
+
+    public RedirectView redirectToSpotifyAuthorization() {
+        String spotifyAuthUrl = "https://accounts.spotify.com/authorize?client_id=" + clientId +
+                "&response_type=code&redirect_uri=" + redirectUri + "&scope=user-read-private%20user-read-email user-library-read user-follow-read user-read-playback-state app-remote-control user-modify-playback-state";
+        return new RedirectView(spotifyAuthUrl);
+    }
+
+    public ResponseEntity<String> spotifyCallback(String code) {
+        accessToken = exchangeAuthCodeForAccessToken(code);
+        return ResponseEntity.ok("Logged in successfully");
+    }
+
+    @Override
+    public String exchangeAuthCodeForAccessToken(String authorizationCode) {
+        RestTemplate restTemplate = new RestTemplate();
+        HttpHeaders headers = new HttpHeaders();
+        headers.setBasicAuth(clientId, clientSecret);
+        headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
+        String scopes = "user-library-read";
+        String requestBody = "grant_type=authorization_code&code=" + authorizationCode +
+                "&redirect_uri=" + redirectUri + "&scope=" + scopes;
+        HttpEntity<String> request = new HttpEntity<>(requestBody, headers);
+        ResponseEntity<String> response = restTemplate.exchange(
+                "https://accounts.spotify.com/api/token",
+                HttpMethod.POST,
+                request,
+                String.class
+        );
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            JsonNode root = mapper.readTree(response.getBody());
+            return root.path("access_token").asText();
+        } catch (IOException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    @Override
+    public String getSavedAlbums(String accessToken) {
+        RestTemplate restTemplate = new RestTemplate();
+
+        HttpHeaders headers = new HttpHeaders();
```

Comment:
We are repeating the implementation of these calls on each one of the methods, please abstract this method calls, header, and exchange. So it can used in all the methods without duplication of code